src/SharpCompress/Common/Zip/ZipFilePart.cs

```diff
@@ -126,16 +126,11 @@ protected Stream CreateDecompressionStream(Stream stream)
 
         protected Stream GetCryptoStream(Stream plainStream)
         {
-            if ((Header.CompressedSize == 0)
-#if !NO_CRYPTO
- && ((Header.PkwareTraditionalEncryptionData != null)
-                    || (Header.WinzipAesEncryptionData != null)))
-#else
-                && (Header.PkwareTraditionalEncryptionData != null))
-#endif
+            if (Header.CompressedSize == 0)
```

Comment:
I guess this is my bad while this is busting things.

I can't think through the logic at the moment but the `null` checks above where used to see if encryption was being attempted.

I guess that's what you were doing with the password being specified.  Sorry, I guess that needs to be put back as I misread.